src/app/modules/group/components/user-list/user-list.component.ts

```diff
@@ -8,43 +10,48 @@ import { GetGroupMembersService, Member } from '../../http-services/get-group-me
   templateUrl: './user-list.component.html',
   styleUrls: [ './user-list.component.scss' ]
 })
-export class UserListComponent implements OnChanges, OnDestroy {
+export class UserListComponent implements OnInit {
 
   @Input() group? : Group;
   state: 'loading' | 'error' | 'empty' | 'ready' = 'loading';
+  currentSort: string[] = [];
 
   members: Member[] = [];
 
-  private subscription?: Subscription;
+  private dataFetching = new BehaviorSubject<'Reload'|'Sort'>('Reload');
 
   constructor(private getGroupMembersService: GetGroupMembersService) { }
 
-  ngOnChanges(_changes: SimpleChanges): void {
-    this.reloadData();
+  ngOnInit(): void {
+    this.dataFetching.pipe(
+      switchMap(mode => {
+        if (this.group) {
+          if (mode === 'Reload')
+            this.state = 'loading';
+
+          return this.getGroupMembersService.getGroupMembers(this.group.id, this.currentSort);
+        } else {
+          throw new Error('group is null');
+        }
+      })
+    ).subscribe(
+      members => {
+        this.members = members;
+
+        if (this.members.length === 0) this.state = 'empty';
```

Comment:
I don't find very relevant to have an "empty" state in this class, you can just check that `members.length === 0` in the template. 
Also the primeng table support the case where it is empty I think... yes there is this "emptyMessageTemplate" that could be used for that, would be better than reimplementing our own logic.

Reply:
fixed